src/lib/crypto/kimchi_bindings/stubs/src/pasta_fp_plonk_proof.rs

```diff
@@ -312,58 +315,89 @@ pub fn caml_pasta_fp_plonk_proof_example_with_foreign_field_mul(
     let right_input = rng.gen_biguint_range(&BigUint::zero(), &foreign_field_modulus);
 
     // Compute multiplication witness
-    let (mut witness, external_checks) =
+    let (mut witness, mut external_checks) =
         foreign_field_mul::witness::create(&left_input, &right_input, &foreign_field_modulus);
 
-    // Bound addition for multiplication result
-    CircuitGate::extend_single_ffadd(
-        &mut gates,
-        &mut next_row,
-        FFOps::Add,
-        &foreign_field_modulus,
-    );
-    gates.connect_cell_pair((1, 0), (2, 0));
-    gates.connect_cell_pair((1, 1), (2, 1));
-    gates.connect_cell_pair((1, 2), (2, 2));
-    external_checks
-        .extend_witness_bound_addition(&mut witness, &foreign_field_modulus.to_field_limbs());
+    // Result compact-multi-range-check
+    CircuitGate::extend_compact_multi_range_check(&mut gates, &mut next_row);
+    gates.connect_cell_pair((1, 0), (4, 1)); // remainder01
+    gates.connect_cell_pair((1, 1), (2, 0)); // remainder2
+    external_checks.extend_witness_compact_multi_range_checks(&mut witness);
```

Comment:
Also perhaps compact MRC witness generation should be moved into `foreign_field_mul::witness::create`  since it's now always required immediately after.